linode/manager
# feat: [M3-7160] - Add AGLB "Edit Certificate" drawer

## Description 📝
Add an edit drawer for AGLB certificates.

## Major Changes 🔄
- Creates the Edit Certificate drawer with dynamic fields depending on certificate type
- Adds the useLoadBalancerCertificateMutation
- Adds a unit test for `EditCertificateDrawer` component
- Updates the MSW `PUT` endpoint for `*/v4beta/aglb/:id/certificates/:certId` to return a `downstream` cert in addition to the existing `ca` certs.

## Preview 📷
| Service Target Certificates  | TLS Certificates   |
| ------- | ------- |
|  ![Screenshot 2023-10-05 at 3 30 54 PM](https://github.com/linode/manager/assets/114685994/8a9f3bf7-d7e3-4d1c-83d9-5c702f2a4683) | ![Screenshot 2023-10-05 at 3 30 38 PM](https://github.com/linode/manager/assets/114685994/8bc5e7e0-6c44-48d3-b742-874ba2526012) |

## How to test 🧪
1. **How to setup test environment?**
- Check out this PR.
- `yarn dev`
- Ensure AGLB feature flag is **on** and MSW is **enabled**.
2. **How to verify changes?**
- Go to http://localhost:3000/loadbalancers/0/certificates/downstream.
- Click on the Edit button in the action menu of the `tls-certificate`. Confirm that the edit drawer matches the mocks above for a TLS certificate.
- Click on the Edit button in the action menu of any other certificate (they're all of type `ca`). Confirm that the edit drawer matches the mocks above for a Service Target certificate.
- Confirm that a `PUT` request is sent by the MSW when the Update Certificate button is clicked.
- Confirm that no `PUT` request is sent by the MSW when the drawer is Canceled or closed out of.
- Confirm form validation:
   - If a user edits the `label` of either certificate type, the form submits without error.
   - If a user edits the `certificate` of a Service Target (`ca`) cert, the form submits without error.
   -  If a user edits the `certificate` of a TLS (`downstream`) cert and submits without a key, they should receive a validation error upon submission  indicating `Private Key is required` in that case.
4. **How to run Unit or E2E tests?**
Unit:
```
yarn test EditCertificateDrawer
```
E2E:
Will be completed in M3-7134, since there was an existing ticket created for this.
